reV/handlers/multi_year.py

```diff
@@ -126,15 +128,53 @@ def _copy_dset(self, source_h5, dset, meta=None, pass_through=False):
             self._create_dset(dset_out, ds_data.shape, ds_dtype,
                               chunks=ds_chunks, attrs=ds_attrs, data=ds_data)
 
+    @staticmethod
+    def parse_source_files_pattern(source_files):
+        """Parse a source_files pattern that can be either an explicit list of
+        source files or a unix-style /filepath/pattern*.h5 and either way
+        return a list of explicit filepaths.
+
+        Parameters
+        ----------
+        source_files : list | str
+            List of .h5 files to collect datasets from. This can also be a
+            unix-style /filepath/pattern*.h5 to find .h5 files to collect,
+            however all resulting files must be .h5 otherwise an exception will
+            be raised. NOTE: .h5 file names much indicate the year the data
+            pertains to
+
+        Returns
+        -------
+        source_files : list
+            List of .h5 filepaths.
+        """
+
+        if isinstance(source_files, (list, tuple)):
```

Comment:
Sorry if I am being dense, but is this check necessary? Does it add some functionality that isn't immediately obvious?

Reply:
Not sure what i was thinking haha. Pretty sure i thought there was going to be a type exception downstream but looks like i never added that. Changing this to an exception on this condition:
```
        elif not isinstance(source_files, (list, tuple)):
            msg = ('Cannot recognize source_files type: {} {}'
                   .format(source_files, type(source_files)))
            logger.error(msg)
            raise TypeError(msg)

Reply:
LGTM